### PR TITLE
fix: remove example recipe.yaml lint

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

Right now we don't have recipe.yaml in our example folder, but we lint for it. 
This lint should be removed.

Fixes https://github.com/conda-forge/staged-recipes/issues/27242

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
Why not just add an example to stages recipes? Otherwise we'll have to add back this lint later.

**Comment by baszalmstra** [comment]
There are [two examples in staged recipes now](https://github.com/conda-forge/staged-recipes/tree/main/recipes), does this properly handle both cases?

**Comment by zklaus** [comment]
The example `recipe.yaml` is in a different folder, namely `example-new-recipe`.

**Comment by beckermr** [comment]
Let's put it in the same folder or adjust the path once in smithy.

**Comment by nichmor** [comment]
My initial thought was that right now we disabled publishing on main channel and to not to confuse our users that there is recipe.yaml as an example but you cannot publish it on the main channel, I decided to remove the lint temporarily.

but I agree with you, let me move it to the same folder

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -483,19 +483,18 @@ def run_conda_forge_specific(
 
     # 4: Do not delete example recipe
     if is_staged_recipes and recipe_dir is not None:
-        for recipe_name in ("meta.yaml", "recipe.yaml"):
-            example_fname = os.path.abspath(
-                os.path.join(recipe_dir, "..", "example", recipe_name)
-            )
```
Comment (jaimergp) [inline conda_smithy/lint_recipe.py:489]: Why not this instead?

```python
for example_path in (("example", "meta.yaml"), ("example-v1", "recipe.yaml")):
    example_fname = os.path.abspath(os.path.join(recipe_dir, "..", *example_path))
```

And then adjust the f-string in the error msg to point to `os.path.join("recipes", *example_path)`?

**Comment by jaimergp** [comment]
Also needs a `news/` item.

**Comment by isuruf** [comment]
Since we have https://github.com/conda-forge/staged-recipes/blob/main/.github/workflows/do_not_edit_example.yml do we even need this lint?

**Comment by beckermr** [comment]
Agreed @isuruf. We should remove to help make smithy more stable and independent of staged-recipes.

**Comment by beckermr** [comment]
See https://github.com/conda-forge/conda-smithy/pull/2038

**Pull request closed without merging**